### PR TITLE
Update Interface for grouping

### PR DESCRIPTION
**Pull request by cptbtptp01**
current interface:
image

Issue:
- #7 
- #32 

Key updates:
- able to handle cluster with/without groups
- added accessibility related annotation to improve screen reader capability 
- update groupItems to return json obj, in order to provide dynamic output in groupingList.tsx

TODOs:
- coordinate with @koolgax99 to update title generated by gpt
- discuss with @koolgax99 and @z-q-ying about how to display each item's content (at element level, current displaying each item's content in a new line.)

### CODE REVIEW

**Comment by cptbtptp01** [comment]
Current is handled by the button. My concern is if we auto the updates, what if screen reader does not finish current task, and the grouping is updated?


------------------ Original ------------------
From: Nihar Sanda ***@***.***&gt;
Date: Fri, Mar 8, 2024 1:57 PM
To: cptbtptp01/ideation-accessibilty ***@***.***&gt;
Cc: Huiru ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [cptbtptp01/ideation-accessibilty] Update Interface for grouping(PR #61)





 
Will the grouping update automatically? @cptbtptp01

File: src/grouping.tsx [hunk 1]
```diff
@@ -38,7 +38,8 @@ export async function groupItems() {
   processAllItems(jsonObject);
 
   // Conver to a string without modifying any properties and with an indentation of 2 spaces
-  return JSON.stringify(jsonObject, null, 2);
+  // return JSON.stringify(jsonObject, null, 2);
+  return jsonObject;
```
Comment (z-q-ying) [inline src/grouping.tsx:42]: Just curious, what will be the difference for returning a json object versus a json string?

**Review by z-q-ying — APPROVED** [review]
The updated interface looks good to me! Fine with using the button to refresh the button too. These look like a very decent milestone~

**Comment by cptbtptp01** [comment]
If return a string, in groupingList I tried to use JSON.parse, I kept getting error in the console and cannot debug on my end. Something like source map not found… then after change to json. Not sure if Nihar was getting similar error.

------------------ Original ------------------
From: z-q-ying ***@***.***&gt;
Date: Fri, Mar 8, 2024 5:17 PM
To: cptbtptp01/ideation-accessibilty ***@***.***&gt;
Cc: Huiru ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [cptbtptp01/ideation-accessibilty] Update Interface for grouping(PR #61)





 
@z-q-ying commented on this pull request.
 
 
In src/grouping.tsx:
 &gt; @@ -38,7 +38,8 @@ export async function groupItems() {    processAllItems(jsonObject);      // Conver to a string without modifying any properties and with an indentation of 2 spaces -  return JSON.stringify(jsonObject, null, 2); +  // return JSON.stringify(jsonObject, null, 2); +  return jsonObject;  
Just curious, what will be the difference for returning a json object versus a json string?

**Comment by z-q-ying** [comment]
I see, thanks for the explanation! Indeed sounds better to return json object in this case, as it's convenient to index while being easy to convert to str whenever needed!

**Pull request merged**